packages/workbox-build/src/lib/bundle.js

```diff
@@ -0,0 +1,109 @@
+/*
+  Copyright 2018 Google LLC
+
+  Use of this source code is governed by an MIT-style
+  license that can be found in the LICENSE file or at
+  https://opensource.org/licenses/MIT.
+*/
+
+const {rollup} = require('rollup');
+const {terser} = require('rollup-plugin-terser');
+const {writeFile} = require('fs-extra');
+const babel = require('rollup-plugin-babel');
+const loadz0r = require('rollup-plugin-loadz0r');
+const path = require('path');
+const replace = require('rollup-plugin-replace');
+const resolve = require('rollup-plugin-node-resolve');
+const tempy = require('tempy');
+
+module.exports = async ({
+  babelPresetEnvTargets,
+  inlineWorkboxRuntime,
+  mode,
+  sourcemap,
+  swDest,
+  unbundledCode,
+}) => {
+  // We need to write this to the "real" file system, as Rollup won't read from
+  // a custom file system.
+  const temporaryFile = tempy.file({name: path.basename(swDest)});
+  await writeFile(temporaryFile, unbundledCode);
+
+  const plugins = [
+    resolve(),
+    replace({'process.env.NODE_ENV': JSON.stringify(mode)}),
+    babel({
+      presets: [['@babel/preset-env', {
+        targets: {
+          browsers: babelPresetEnvTargets,
+        },
+        loose: true,
+      }]],
+    }),
+  ];
+
+  if (mode === 'production') {
+    plugins.push(terser({
+      mangle: {
+        toplevel: true,
+        properties: {
+          regex: /(^_|_$)/,
+        },
+      },
+    }));
+  }
+
+  const rollupConfig = {
+    plugins,
+    input: temporaryFile,
+    treeshake: {
+      moduleSideEffects: false,
```

Comment:
Didn't you determine that this removes the versioning code?

Reply:
Yeah, it does.

I think I'll prevent the "too many versioning strings in the final bundle" issue by conditionally toggling the various `import` statements in the service worker source based on whether they're actually needed, instead of relying on dead code elimination.

Reply:
I'm now dynamically generating the list of `import` statements based on actual usage. It's a little funky, but it means that we aren't reliant on dead code elimination.